src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -178,7 +177,7 @@ public unsafe void CloseSocket()
                     WriteLog.To.Sync.I(Tag, "Closing socket normally due to request from LiteCore");
                     ReleaseSocket(new C4Error(0, 0));
                 });
-            });
+            }).ContinueWith(t=> ResetConnections());
```

Comment:
Sorry, I don't know .NET. Does this get executed before or after `ReleaseSocket(new C4Error(0, 0));`?

Reply:
Reset first and then release Socket.

Reply:
Let me reverse the order.

Reply:
I thought @pasin said `ResetConnections` should be last....if it needs to happen *before* the logic inside of the c4Queue then a different approach is needed.